dev/devicelab/lib/tasks/gallery.dart

```diff
@@ -251,25 +251,44 @@ class GalleryTransitionBuildTest extends BuildTestTask {
 
   @override
   void copyArtifacts() {
-    if(applicationBinaryPath != null) {
+    if (applicationBinaryPath == null) {
+      return;
+    }
+    if (deviceOperatingSystem == DeviceOperatingSystem.android) {
       copy(
         file('${galleryDirectory.path}/build/app/outputs/flutter-apk/app-profile.apk'),
         Directory(applicationBinaryPath!),
       );
+    } else if (deviceOperatingSystem == DeviceOperatingSystem.ios) {
+      recursiveCopy(
+        Directory('${galleryDirectory.path}/build/ios/iphoneos'),
```

Comment:
And copying only this directory prevents a rebuild?

Reply:
Yes. Validated using LED runs. Example: https://luci-milo.appspot.com/raw/build/logs.chromium.org/flutter/led/keyonghan_google.com/a43684afc067cba831ca3268c9621008cae1b916634886dada4c5c4a8c3d3587/+/build.proto, which skipped the `build` step, and ran test based on earlier built artifact.